packages/core/lib/command-utils.js

```diff
@@ -48,7 +48,12 @@ const getCommand = ({ inputStrings, options, noAliases }) => {
   }
 
   if (chosenCommand == null) {
-    return null;
+    //error out and exit when we cannot find any command that matches.
+    let err = new Error(
+      `\`truffle ${firstInputString}\` is not a valid truffle command. Please see \`truffle help\` for available commands.`
+    );
+    console.log(err.message);
+    process.exit(err);
```

Comment:
in general you don't call `process.exit` with an `Error` object. you should call it with an exit code: `0` if everything went successfully and `1` if not. you can check out the docs on this [here](https://nodejs.dev/learn/how-to-exit-from-a-nodejs-program)

Reply:
Note: this file already uses `process.exit` in other places.